functions/_utils/getNotificationRecipients.ts

```diff
@@ -0,0 +1,34 @@
+import { gql } from '@gql'
+import { adminRequest } from '@utils/adminRequest'
+import { RouteError } from '@utils/route'
+import { TriggerRecipientsPayload } from '@novu/node'
+
+export async function getNotificationRecipients(recipientMemberIds: string[]) {
+  const recipientsResult = await adminRequest(GET_RECIPIENTS, {
+    memberIds: recipientMemberIds,
+  })
+  if (!recipientsResult) throw new RouteError(404, 'Member not found')
+
+  const recipients = recipientsResult.member
+    .map((member) =>
+      member.user
+        ? {
+            subscriberId: member.user.id!,
+            email: member.user.email!,
```

Comment:
A-t-on besoin de l'email étant donné qu'on a déjà identifié le user auprès de Novu ?

Reply:
Je l'avais mis au cas où en sécurité
Si à tout hasard l'email venait à ne plus être chez Novu au moins on le passe nous dans tous les cas 
Mais peut être qu'on ne préfère pas le passer, je te laisse me dire

Reply:
Tu sais si Novu crée le user automatiquement (ou tout du moins fonctionne) si on envoie une notif sans avoir créé le user avant ?

Reply:
"When the trigger is called, Novu performs an upsert command, which either creates a subscriber with the specified payload, or updates the existing subscriber with the passed information." (dans le doc de Novu)
Donc il semble bien créer un subscriber à la volée s'il n'existe pas (il faudrait donc passer email et locale pour le créer avec ces data).
Par contre côté front à bien noter que le centre de notif' se base sur le subscriberId
- s'il n'est pas créer/identifier, le centre est en  400 à l'initialisation car le subscriberId qu'on passe n'existe pas encore chez Novu 
- le centre conserve tout de même son display initial, la seule chose est qu'il reste en loader infini dans ce cas
Je n'avais pas pensé à cette façon de faire mais en effet à voir si ça ne nous pose pas de problème de faire comme ça
Ici ça voudrait donc dire : 
- le subscriber sera créer lorsqu'on lui enverra sa première notif'
- tant qu'une notif au moins n'a pas été envoyée au user, le centre de notif sera en 400 avec son loader infini
- il faudrait donc peut être voir à changer avec un texte "Non connecté" ou quelque chose du genre si possible ?